Formula/texlive.rb

```diff
@@ -0,0 +1,499 @@
+      "--without-x"
+    else
+      # Make sure xdvi uses xaw, even if motif is available
+      "--with-xdvi-x-toolkit=xaw"
+    end
+
+    system "./configure", *args
+    system "make"
+    system "make", "install"
+    system "make", "texlinks"
+
+    # Create tlmgr config file.  This file limits the actions that the user
+    # can perform in 'system' mode, which would write to the cellar.  'tlmgr' should
+    # be used with --usermode whenever possible.
+    (share/"texmf-config/tlmgr/config").write <<~EOS
+      allowed-actions=candidates,check,dump-tlpdb,help,info,list,print-platform,print-platform-info,search,show,version,init-usertree
+    EOS
+
+    # Delete some binaries that are provided by existing formulae as newer versions.
+    rm bin/"latexindent" # provided by latexindent formula
+    rm bin/"latexdiff" # provided by latexdiff formula
+    rm bin/"latexdiff-vc" # provided by latexdiff formula
+    rm bin/"latexrevise" # provided by latexdiff formula
+
+    # Wrap some Perl scripts in an env script so that they can find dependencies
+    env_script_files = [
+      "crossrefware/bbl2bib.pl",
+      "crossrefware/bibdoiadd.pl",
+      "crossrefware/bibmradd.pl",
+      "crossrefware/biburl2doi.pl",
+      "crossrefware/bibzbladd.pl",
+      "crossrefware/ltx2crossrefxml.pl",
+      "ctan-o-mat/ctan-o-mat.pl",
+      "ctanify/ctanify",
+      "ctanupload/ctanupload.pl",
+      "exceltex/exceltex",
+      "latex-git-log/latex-git-log",
+      "pax/pdfannotextractor.pl",
+      "ptex-fontmaps/kanji-fontmap-creator.pl",
+      "purifyeps/purifyeps",
+      "svn-multi/svn-multi.pl",
+      "texdoctk/texdoctk.pl",
+      "ulqda/ulqda.pl",
+    ]
+
+    env_script_files.each do |perl_script|
+      bin_name = File.basename(perl_script, ".pl")
+      rm bin/bin_name
+      (bin/bin_name).write_env_script(share/"texmf-dist/scripts"/perl_script, PERL5LIB: ENV["PERL5LIB"])
+    end
+
+    # Wrap some Python scripts so they can find dependencies.
+    python_path = libexec/Language::Python.site_packages("python3")
+    ENV.prepend_path "PYTHONPATH", python_path
+    (bin/"pygmentex").write_env_script(share/"texmf-dist/scripts/pygmentex/pygmentex.py",
+      PYTHONPATH: ENV["PYTHONPATH"])
+    (bin/"pythontex").write_env_script(share/"texmf-dist/scripts/pythontex/pythontex.py",
+      PYTHONPATH: ENV["PYTHONPATH"])
+
+    # Rewrite shebangs in some Python scripts so they use brewed Python.
+    python_shebang_rewrites = [
+      "latex-papersize/latex-papersize.py",
+      "lilyglyphs/lily-glyph-commands.py",
+      "lilyglyphs/lily-image-commands.py",
+      "lilyglyphs/lily-rebuild-pdfs.py",
+      "texliveonfly/texliveonfly.py",
+    ]
+
+    python_shebang_rewrites.each do |python_script|
+      rewrite_shebang detected_python_shebang, share/"texmf-dist/scripts"/python_script
+    end
+
+    # Delete ebong on Linux, because it requires Python 2
+    rm bin/"ebong" if OS.linux?
+
+    # Initialize texlive environment
```

Comment:
Is this okay being overwritten? Or should it be in postinstall?

Reply:
I think this should be in the postinstall.  Good catch!

Reply:
a question: is postinstall run even when using bottle, or only at build time? These commands take considerable time.

Reply:
Yes, it is run when using a bottle, and while it is slow I think it makes sense to run this in postinstall.  The extraction of the bottle itself is probably much slower than these post install steps given the size and number of files, so I don't know if having to run this will really add that much to the install time.  If I'm wrong and this is the slowest part when pouring a bottle, let me know and we can revisit this.

Reply:
Oh no, it does not take much time like compile and download/copy resources.
You can put it on postinstall.

Reply:
Some subsequent testing revealed that putting this in the postinstall actually doesn't work, because it needs write access to the keg.  Everything that is being done by those tools should be treated as immutable afterward, so it's okay for us to run this during the install block.